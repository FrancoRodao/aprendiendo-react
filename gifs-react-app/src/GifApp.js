import React, {useState} from 'react'
import AddCategory from "./components/AddCategory";
import { GifGrid} from './components/GifGrid';

export default function GifApp({ initialCategories = [] }) {
    
    const [categories, setCategories] = useState(initialCategories)
    
    return (
        <>
            <h2>Gif App</h2>
            <hr/>
            <AddCategory setCategories={setCategories}></AddCategory>
            <ol>
                {
                    categories.map(category => <GifGrid category={category} key={category}></GifGrid>)
                }
            </ol>
        </>
    )
}
