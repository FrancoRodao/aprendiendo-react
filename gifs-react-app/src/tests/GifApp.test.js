import React from 'react'
import { shallow } from "enzyme";
import GifApp from "../GifApp";

describe('test GifApp component "app container" )', () => {
    const intialCategories = ["testing"]
    let wrapper = shallow(<GifApp initialCategories={intialCategories}></GifApp>)

    test('should display snapshot correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })
    
    test('should display GifGrids components', () => {
        expect(wrapper.find("GifGrid").length).toBe(intialCategories.length)
    })
    
    
})
