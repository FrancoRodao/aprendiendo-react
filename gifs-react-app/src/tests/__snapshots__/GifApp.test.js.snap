// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test GifApp component "app container" ) should display snapshot correctly 1`] = `
<Fragment>
  <h2>
    Gif App
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="testing"
      key="testing"
    />
  </ol>
</Fragment>
`;
